### PR TITLE
Added neural network functionality

### PR DESCRIPTION
**Pull request by samysspace**
Added a neural network adaptor to adaptors.py using a Scikit-Learn Keras wrapper with a full test suite (all 4 tests pass) in test_nn.py in the tests folder. The neural network is getting pretty good results for both regression and classification in a matter of seconds so this might be a cool feature to add in.

### CODE REVIEW

File: automatminer/automl/adaptors.py [hunk 1]
```diff
@@ -226,6 +228,129 @@ def predict(self, df, target):
...
+        Returns:
+            (pandas.DataFrame): The argument dataframe plus a column containing
+                the predictions of the target.
+
+        """
+        if target != self.fitted_target:
+            raise AutomatminerErrorError("Argument dataframe target {} is different from"
+                                " the fitted dataframe target! {}"
+                                "".format(target, self.fitted_target))
+        elif not self.is_fit:
```
Comment (ardunn) [inline automatminer/automl/adaptors.py]: This is_fit check here is superflous. Check @check_fitted decorator for details. We'll be removing these similar lines from TPOTAdaptor

File: automatminer/automl/adaptors.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
     regression_or_classification
 from automatminer.base import AutoMLAdaptor, LoggableMixin
 
+from automatminer.automl.keras_wrapper import NnWrapper
+
 __authors__ = ['Alex Dunn <ardunn@lbl.gov'
```
Comment (ardunn) [inline automatminer/automl/adaptors.py]: add yourself to the authors! @samysspace

File: automatminer/automl/adaptors.py [hunk 3]
```diff
@@ -18,6 +18,8 @@
     regression_or_classification
 from automatminer.base import AutoMLAdaptor, LoggableMixin
 
+from automatminer.automl.keras_wrapper import NnWrapper
```
Comment (ardunn) [inline automatminer/automl/adaptors.py]: remove the space between imports? (or this might be github making it look like this?)

File: automatminer/automl/adaptors.py [hunk 4]
```diff
@@ -226,6 +228,129 @@ def predict(self, df, target):
...
+        Returns:
+            NeuralNetworkAdaptor(self)
+
+        """
+        y = df[target]
+        X = df.drop(columns=target)
+
+        # Determine learning type based on whether classification or regression
+        self.mode = regression_or_classification(df[target])
+        if self.mode == "classification" or self.mode == "regression":
```
Comment (ardunn) [inline automatminer/automl/adaptors.py]: this "if" check is not needed. `regression_or_classification" will return either regression or classification, or throw an error otherwise.

File: automatminer/automl/tests/test_nn.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import pandas as pd
+from sklearn.metrics import r2_score, f1_score
+
+from automatminer.automl.adaptors import NeuralNetworkAdaptor
+from automatminer.utils.package_tools import AutomatminerError
+
+__author__ = ['Qi Wang <qwang3@lbl.gov>', 'Alex Dunn <ardunn@lbl.gov>']
+
+
+@unittest.skipIf("CI" in os.environ.keys(), "Test too intensive for CircleCI.")
```
Comment (ardunn) [inline automatminer/automl/tests/test_nn.py]: Are these tests actually too intensive for CircleCI? If so it's fine to skip for now. The TPOT ones are excluded because tpot isn't that great with handling memory or the model search space, even when using debugging settings.

File: requirements-optional.txt [hunk 6]
```diff
@@ -1 +1,3 @@
-skater==1.1.2
\ No newline at end of file
+skater==1.1.2
+Keras==2.2.4
```
Comment (ardunn) [inline requirements-optional.txt]: I think it's ok to actually add keras/tensorflow requirements to the requirements.txt, not the optional reqs. I mostly put skater in here because its build is failing on its own repo and its not pip-installable (wordcloud requirement build fails)

File: automatminer/automl/tests/test_nn.py (hunk 5 above)
Comment (ardunn) [reply automatminer/automl/tests/test_nn.py]: also, add yourself to the authors again!

**Comment by samysspace** [comment]
It looks like I removed all the automated functionality when I was testing the accuracy of certain individual models and accidentally forgot to add it back. I'll fix that (along with all the other issues) presently.

**Comment by ardunn** [comment]
Hey samy, you can just keep this PR open until the issues are fixed. Take another week with it and just push to your branch

**Comment by samysspace** [comment]
Unfortunately, I made this PR by creating a branch from the main repo instead of forking the repo so I lost access to it when I was changed to view-only rights. I created a fork and have pushed my updates to another pull request (where I fixed all the issues mentioned above): https://github.com/hackingmaterials/automatminer/pull/153. This one should hopefully be the permanent pull request.
